File: C/Assignment10/Program5.c
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                                                          //
// File name :   Program5.c                                                                                                 //
// Description : Program which accept number from user & return the differenc between Summation of even & odd digits        //  
// Author :      Pallavi Omprakash Malewar                                                                                  //
// Date :        20/06/2025                                                                                                 //
//                                                                                                                          //
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

/*
Input : 2395
Output : -15

Input : 1018
Output : 6

Input : 8440
Output : 16

Input : 5733
Output : -18
*/

#include<stdio.h>

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                                                          //
// Description   : Function accepts integer and returns difference between summation of even and odd digits.                //
// Function Name : CountDiff                                                                                                //
// Input         : int iNo                                                                                                  //
// Output        : int (Difference between sum of even and odd digits)                                                      //
// Author        : Pallavi Omprakash Malewar                                                                                //
// Date          : 20/06/2025                                                                                               //
//                                                                                                                          //
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

int CountDiff(int iNo)
{
    int iDigit = 0 , iSumE = 0 , iSumO = 0 , iDiff = 0;

    while(iNo != 0)
    {
        iDigit = iNo % 10;
        if((iDigit % 2) == 0)
        {
            iSumE = iSumE + iDigit;
        }
        else
        {
            iSumO = iSumO + iDigit;
        }
        iNo = iNo / 10;
    }
    iDiff = iSumE - iSumO;
    return iDiff;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter Number : ");
    scanf("%d",&iValue);

    iRet = CountDiff(iValue);

    printf("Difference between summation of even & odd digits : %d\n",iRet);

    return 0;
}

File: JAVA/Assignment10/Program5.java
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// Problem Statement : Write a program which accept number from user and return diff   //
//                     between summation of even digits and summation odd digits.      //
// Author : Pallavi Omprakash Malewar                                                  //
// Date : 30 Jun 2026                                                                  //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////

/*

    Input: 2395
    Output : 270

    Input: 1018
    Output : 8

    Input: 9440
    Output : 144

    Input: 922432
    Output : 864

*/


import java.util.Scanner;

class Program5
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter the Number : ");
        int No = sobj.nextInt();

        int diff = CountDiff(No);

        System.out.println(diff);
    }

    public static int CountDiff(int No)
    {
        int iDigit = 0, EvenSum = 0, OddSum = 0;

        if(No == 0) return 0;
        if(No < 0) No = -No;

        while(No != 0)
        {
            iDigit = No % 10;
            if(iDigit % 2 == 0) EvenSum += iDigit;
            else OddSum += iDigit;

            No /= 10;
        }
        int diff = EvenSum - OddSum;
        return diff;
    }
}